share/lib/python/neuron/tests/test_neuron.py

```diff
@@ -103,6 +103,25 @@ def testSectionArgOrder(self):
         soma = h.Section('soma')
         assert soma.name() == 'soma'
 
+    def testSectionCell(self):
+        """ Section.cell() internally referenced as weakref."""
+        err = -1
+        try:
+            soma = h.Section(cell="foo", name="soma")
+            err = 1
+        except:
+            err = 0
+        assert err == 0
+        class Cell():
+            def __str__(self):
+                return "hello"
+        c = Cell()
+        soma = h.Section(cell=c, name="soma")
+        assert soma.name() == "hello.soma"
+        assert soma.cell() == c
+        del c
+        assert soma.cell() == None
+
```

Comment:
Haven't tested, but from the code it looks like `soma.name()` is still `"hello.soma"` after the cell is deleted. Is that the "right" behavior?

I'm leaning towards it's the right thing, but I wanted to raise the question.

Reply:
I was perplexed by this case also. It represents an earlier misuse of the  ```cell=...``` argument in that the section is supposed to be referenced by the container and destroyed when the container is destroyed. I suppose it would be possible to invalidate the section in the case where the section lives outside the cell but I don't think that is the correct behavior either.
The alternative of removing that part of the name so that it suddenly appears to be a section at the "top level" I think is inappropriate.  Augmenting the cell part of the name so that it indicates  the cell has been destroyed I don't think is worthwhile. Another possibility is to generate a warning or an error if this "misuse" happens. As the name of a Python Section is not entirely substantive (merely a label for gui display) I tend toward leaving it as is.